ballerina/client.bal

```diff
@@ -30,6 +30,18 @@ public type Client client object {
     remote isolated function query(string|ParameterizedQuery sqlQuery, typedesc<record {}> rowType = <>)
     returns stream <rowType, Error?>;
 
+    # Queries the database with the provided query and returns the first row as a record if the expected return type is
+    # a record. If the expected return type is not a record, then a single value is returned.
+    #
+    # + sqlQuery - The query, which needs to be executed as a `string` or  an `sql:ParameterizedQuery` when the SQL
+    #               query has params to be passed in
+    # + returnType - The `typedesc` of the record/type that should be returned as a result. If this is not provided, the
+    #                default column names/type of the query result set will be used
+    # + return - Result in the type of `returnType`. If the `returnType` is not provided, the column names/type of
+    #               the query are used
+    remote isolated function queryRow(string|ParameterizedQuery sqlQuery, typedesc<any> returnType = <>)
```

Comment:
If the query returns multiple rows when only a single row is expected, only the first row is returned.

Errors are thrown in the following cases,
- Query retrieves an empty resultSet
- Query retrieves multiple rows when the expected return type is a value
- Query retrieves multiple columns when the expected return type is a v

Reply:
For the first scenario, shall we return null at the native java level, user can assign it to a nilable variable.
For the second scenario, if a query returns multiple rows, we only return the first row and discard others.

Reply:
Fixed in 5eeb372

First scenario was discussed to be expected behaviour. Will return a `NoRowsError`.
Second scenario now returns just the first row's value.